Enable compiler warnings

By default Maven generates only a non-helpful warning for deprecated Java code:

```
[INFO] --- maven-compiler-plugin:3.11.0:testCompile (default-testCompile) @ exasol-testcontainers ---
[INFO] Changes detected - recompiling the module! :dependency
[INFO] Compiling 72 source files with javac [debug target 11] to target/test-classes
[INFO] /home/runner/work/exasol-testcontainers/exasol-testcontainers/src/test/java/com/exasol/clusterlogs/MappedClusterLogsIT.java: /home/runner/work/exasol-testcontainers/exasol-testcontainers/src/test/java/com/exasol/clusterlogs/MappedClusterLogsIT.java uses or overrides a deprecated API.
[INFO] /home/runner/work/exasol-testcontainers/exasol-testcontainers/src/test/java/com/exasol/clusterlogs/MappedClusterLogsIT.java: Recompile with -Xlint:deprecation for details.
[INFO] 
```

To fix this, we need to add option to the compiler plugin:

```xml
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.11.0</version>
                <configuration>
                    <source>${java.version}</source>
                    <target>${java.version}</target>
                    <showWarnings>true</showWarnings>
                    <compilerArgs>
                        <arg>-Xlint:all</arg>
                    </compilerArgs>
                </configuration>
            </plugin>
```

Note: `<arg>-Werror</arg>` would let the build fail with the following error:

```
[INFO] --- maven-compiler-plugin:3.11.0:compile (default-compile) @ exasol-testcontainers ---
[INFO] Changes detected - recompiling the module! :source
[INFO] Compiling 71 source files with javac [debug target 11] to target/classes
[INFO] -------------------------------------------------------------
[WARNING] COMPILATION WARNING : 
[INFO] -------------------------------------------------------------
[WARNING] system modules path not set in conjunction with -source 11
[INFO] 1 warning
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] warnings found and -Werror specified
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

--- a/dependencies.md
+++ b/dependencies.md
@@ -304,7 +304,7 @@
 [15]: https://www.jqno.nl/equalsverifier
 [16]: https://www.apache.org/licenses/LICENSE-2.0.txt
 [17]: https://github.com/mockito/mockito
-[18]: https://github.com/mockito/mockito/blob/main/LICENSE
+[18]: https://opensource.org/licenses/MIT
 [19]: http://www.slf4j.org
 [20]: http://www.opensource.org/licenses/mit-license.php
 [21]: http://sonarsource.github.io/sonar-scanner-maven/

--- a/doc/changes/changelog.md
+++ b/doc/changes/changelog.md
@@ -1,5 +1,6 @@
 # Changes
 
+* [2.9.16](changes_2.9.16.md)
 * [2.9.15](changes_2.9.15.md)
 * [2.9.14](changes_2.9.14.md)
 * [2.9.13](changes_2.9.13.md)

--- a/doc/changes/changes_2.9.16.md
+++ b/doc/changes/changes_2.9.16.md
@@ -1,0 +1,55 @@
+# Project Keeper 2.9.16, released 2023-??-??
+
+Code name: Support Matrix builds & enable compiler warnings
+
+## Summary
+
+This release enables linter warnings for the Java compiler by adding argument `-Xlint:all`. This will log warnings but the build won't fail.
+
+## Features
+
+* #497: Enabled linter warnings
+
+## Dependency Updates
+
+### Project Keeper Core
+
+#### Compile Dependency Updates
+
+* Updated `com.exasol:project-keeper-shared-model-classes:2.9.15` to `2.9.16`
+
+#### Runtime Dependency Updates
+
+* Updated `com.exasol:project-keeper-java-project-crawler:2.9.15` to `2.9.16`
+
+#### Test Dependency Updates
+
+* Updated `com.exasol:project-keeper-shared-test-setup:2.9.15` to `2.9.16`
+
+### Project Keeper Command Line Interface
+
+#### Compile Dependency Updates
+
+* Updated `com.exasol:project-keeper-core:2.9.15` to `2.9.16`
+
+#### Test Dependency Updates
+
+* Updated `com.exasol:project-keeper-shared-test-setup:2.9.15` to `2.9.16`
+
+### Project Keeper Maven Plugin
+
+#### Compile Dependency Updates
+
+* Updated `com.exasol:project-keeper-core:2.9.15` to `2.9.16`
+
+### Project Keeper Java Project Crawler
+
+#### Compile Dependency Updates
+
+* Updated `com.exasol:project-keeper-shared-model-classes:2.9.15` to `2.9.16`
+
+### Project Keeper Shared Test Setup
+
+#### Compile Dependency Updates
+
+* Updated `com.exasol:project-keeper-shared-model-classes:2.9.15` to `2.9.16`

--- a/doc/changes/changes_2.9.16.md
+++ b/doc/changes/changes_2.9.16.md
@@ -12,6 +12,20 @@ This release enables linter warnings for the Java compiler by adding argument `-
 
 ## Dependency Updates
 
+### Project-Keeper Shared Model Classes
+
+#### Test Dependency Updates
+
+* Updated `nl.jqno.equalsverifier:equalsverifier:3.15.2` to `3.15.3`
+* Updated `org.junit.jupiter:junit-jupiter-engine:5.10.0` to `5.10.1`
+* Updated `org.junit.jupiter:junit-jupiter-params:5.10.0` to `5.10.1`
+* Updated `org.mockito:mockito-core:5.6.0` to `5.7.0`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.0` to `3.6.2`
+* Updated `org.apache.maven.plugins:maven-surefire-plugin:3.1.2` to `3.2.2`
+
 ### Project Keeper Core
 
 #### Compile Dependency Updates
@@ -25,6 +39,16 @@ This release enables linter warnings for the Java compiler by adding argument `-
 #### Test Dependency Updates
 
 * Updated `com.exasol:project-keeper-shared-test-setup:2.9.15` to `2.9.16`
+* Updated `nl.jqno.equalsverifier:equalsverifier:3.15.2` to `3.15.3`
+* Updated `org.junit.jupiter:junit-jupiter-engine:5.10.0` to `5.10.1`
+* Updated `org.junit.jupiter:junit-jupiter-params:5.10.0` to `5.10.1`
+* Updated `org.mockito:mockito-junit-jupiter:5.6.0` to `5.7.0`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-failsafe-plugin:3.1.2` to `3.2.2`
+* Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.0` to `3.6.2`
+* Updated `org.apache.maven.plugins:maven-surefire-plugin:3.1.2` to `3.2.2`
 
 ### Project Keeper Command Line Interface
 
@@ -35,6 +59,14 @@ This release enables linter warnings for the Java compiler by adding argument `-
 #### Test Dependency Updates
 
 * Updated `com.exasol:project-keeper-shared-test-setup:2.9.15` to `2.9.16`
+* Updated `org.junit.jupiter:junit-jupiter-engine:5.10.0` to `5.10.1`
+* Updated `org.junit.jupiter:junit-jupiter-params:5.10.0` to `5.10.1`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-failsafe-plugin:3.1.2` to `3.2.2`
+* Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.0` to `3.6.2`
+* Updated `org.apache.maven.plugins:maven-surefire-plugin:3.1.2` to `3.2.2`
 
 ### Project Keeper Maven Plugin
 
@@ -42,14 +74,47 @@ This release enables linter warnings for the Java compiler by adding argument `-
 
 * Updated `com.exasol:project-keeper-core:2.9.15` to `2.9.16`
 
+#### Test Dependency Updates
+
+* Updated `org.junit.jupiter:junit-jupiter-engine:5.10.0` to `5.10.1`
+* Updated `org.junit.jupiter:junit-jupiter-params:5.10.0` to `5.10.1`
+* Updated `org.mockito:mockito-core:5.6.0` to `5.7.0`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-dependency-plugin:3.6.0` to `3.6.1`
+* Updated `org.apache.maven.plugins:maven-failsafe-plugin:3.1.2` to `3.2.2`
+* Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.0` to `3.6.2`
+* Updated `org.apache.maven.plugins:maven-plugin-plugin:3.10.1` to `3.10.2`
+* Updated `org.apache.maven.plugins:maven-surefire-plugin:3.1.2` to `3.2.2`
+
 ### Project Keeper Java Project Crawler
 
 #### Compile Dependency Updates
 
 * Updated `com.exasol:project-keeper-shared-model-classes:2.9.15` to `2.9.16`
 
+#### Test Dependency Updates
+
+* Updated `org.junit.jupiter:junit-jupiter-engine:5.10.0` to `5.10.1`
+* Updated `org.junit.jupiter:junit-jupiter-params:5.10.0` to `5.10.1`
+* Updated `org.mockito:mockito-core:5.6.0` to `5.7.0`
+* Updated `org.mockito:mockito-junit-jupiter:5.6.0` to `5.7.0`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-dependency-plugin:3.6.0` to `3.6.1`
+* Updated `org.apache.maven.plugins:maven-failsafe-plugin:3.1.2` to `3.2.2`
+* Updated `org.apache.maven.plugins:maven-javadoc-plugin:3.6.0` to `3.6.2`
+* Updated `org.apache.maven.plugins:maven-plugin-plugin:3.10.1` to `3.10.2`
+* Updated `org.apache.maven.plugins:maven-surefire-plugin:3.1.2` to `3.2.2`
+
 ### Project Keeper Shared Test Setup
 
 #### Compile Dependency Updates
 
 * Updated `com.exasol:project-keeper-shared-model-classes:2.9.15` to `2.9.16`
+
+#### Plugin Dependency Updates
+
+* Updated `org.apache.maven.plugins:maven-surefire-plugin:3.1.2` to `3.2.2`

--- a/maven-project-crawler/pk_generated_parent.pom
+++ b/maven-project-crawler/pk_generated_parent.pom
@@ -141,7 +141,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.2.2</version>
                 <configuration>
                     <!-- Set the highest log level for coverage testing, so that we
           have a chance to reach branches in the logging lambdas too. -->
@@ -239,7 +239,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.6.0</version>
+                <version>3.6.2</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>
@@ -281,7 +281,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-dependency-plugin</artifactId>
-                <version>3.6.0</version>
+                <version>3.6.1</version>
                 <executions>
                     <execution>
                         <id>copy-jacoco</id>
@@ -301,10 +301,9 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-failsafe-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.2.2</version>
                 <configuration>
-                    <!-- Set the highest log level for coverage testing, so that we have a chance to reach branches
-          <property>	                      in the logging lambdas too. -->
+                    <!-- Set the highest log level for coverage testing, so that we have a chance to reach branches in the logging lambdas too. -->
                     <argLine>-Djava.util.logging.config.file=src/test/resources/logging.properties ${argLine}</argLine>
                     <systemPropertyVariables>
                         <test.coverage>true</test.coverage>

--- a/maven-project-crawler/pk_generated_parent.pom
+++ b/maven-project-crawler/pk_generated_parent.pom
@@ -71,6 +71,12 @@
                 <configuration>
                     <source>${java.version}</source>
                     <target>${java.version}</target>
+                    <showWarnings>true</showWarnings>
+                    <compilerArgs>
+                        <!-- We turn off `processing` because it complains about every annotation -->
+                        <arg>-Xlint:all,-processing</arg>
+                        <!-- Don't add -Werror as this will let the build fail for linter warnings -->
+                    </compilerArgs>
                 </configuration>
             </plugin>
             <plugin>

--- a/maven-project-crawler/pom.xml
+++ b/maven-project-crawler/pom.xml
@@ -99,7 +99,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-plugin-plugin</artifactId>
-                <version>3.10.1</version>
+                <version>3.10.2</version>
             </plugin>
             <plugin>
                 <groupId>org.basepom.maven</groupId>

--- a/parent-pom/pom.xml
+++ b/parent-pom/pom.xml
@@ -28,7 +28,7 @@
         </repository>
     </distributionManagement>
     <properties>
-        <revision>2.9.15</revision>
+        <revision>2.9.16</revision>
         <maven.version>3.9.5</maven.version>
         <minimum.maven.version>3.6.3</minimum.maven.version>
         <junit.version>5.10.0</junit.version>

--- a/parent-pom/pom.xml
+++ b/parent-pom/pom.xml
@@ -81,6 +81,12 @@
                 <version>3.0.3</version>
             </dependency>
             <dependency>
+                <!-- Upgrade transitive dependency of org.eclipse:yasson to fix CVE-2023-4043 -->
+                <groupId>org.eclipse.parsson</groupId>
+                <artifactId>parsson</artifactId>
+                <version>1.1.5</version>
+            </dependency>
+            <dependency>
                 <groupId>org.apache.maven.plugin-tools</groupId>
                 <artifactId>maven-plugin-annotations</artifactId>
                 <version>3.10.2</version>

--- a/parent-pom/pom.xml
+++ b/parent-pom/pom.xml
@@ -31,9 +31,9 @@
         <revision>2.9.16</revision>
         <maven.version>3.9.5</maven.version>
         <minimum.maven.version>3.6.3</minimum.maven.version>
-        <junit.version>5.10.0</junit.version>
+        <junit.version>5.10.1</junit.version>
         <xmlunit.version>2.9.1</xmlunit.version>
-        <mockito.version>5.6.0</mockito.version>
+        <mockito.version>5.7.0</mockito.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <maven.compiler.source>11</maven.compiler.source>
@@ -83,7 +83,7 @@
             <dependency>
                 <groupId>org.apache.maven.plugin-tools</groupId>
                 <artifactId>maven-plugin-annotations</artifactId>
-                <version>3.10.1</version>
+                <version>3.10.2</version>
                 <scope>provided</scope>
             </dependency>
             <dependency>
@@ -198,7 +198,7 @@
             <dependency>
                 <groupId>nl.jqno.equalsverifier</groupId>
                 <artifactId>equalsverifier</artifactId>
-                <version>3.15.2</version>
+                <version>3.15.3</version>
                 <scope>test</scope>
             </dependency>
             <!-- overriding vulnerable version -->

--- a/project-keeper-cli/pk_generated_parent.pom
+++ b/project-keeper-cli/pk_generated_parent.pom
@@ -63,6 +63,12 @@
                 <configuration>
                     <source>${java.version}</source>
                     <target>${java.version}</target>
+                    <showWarnings>true</showWarnings>
+                    <compilerArgs>
+                        <!-- We turn off `processing` because it complains about every annotation -->
+                        <arg>-Xlint:all,-processing</arg>
+                        <!-- Don't add -Werror as this will let the build fail for linter warnings -->
+                    </compilerArgs>
                 </configuration>
             </plugin>
             <plugin>

--- a/project-keeper-cli/pk_generated_parent.pom
+++ b/project-keeper-cli/pk_generated_parent.pom
@@ -133,7 +133,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.2.2</version>
                 <configuration>
                     <!-- Set the highest log level for coverage testing, so that we
           have a chance to reach branches in the logging lambdas too. -->
@@ -281,7 +281,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.6.0</version>
+                <version>3.6.2</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>
@@ -323,10 +323,9 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-failsafe-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.2.2</version>
                 <configuration>
-                    <!-- Set the highest log level for coverage testing, so that we have a chance to reach branches
-          <property>	                      in the logging lambdas too. -->
+                    <!-- Set the highest log level for coverage testing, so that we have a chance to reach branches in the logging lambdas too. -->
                     <argLine>-Djava.util.logging.config.file=src/test/resources/logging.properties ${argLine}</argLine>
                     <systemPropertyVariables/>
                     <excludedGroups>${test.excludeTags}</excludedGroups>

--- a/project-keeper-maven-plugin/pk_generated_parent.pom
+++ b/project-keeper-maven-plugin/pk_generated_parent.pom
@@ -141,7 +141,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.2.2</version>
                 <configuration>
                     <!-- Set the highest log level for coverage testing, so that we
           have a chance to reach branches in the logging lambdas too. -->
@@ -239,7 +239,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.6.0</version>
+                <version>3.6.2</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>
@@ -281,7 +281,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-dependency-plugin</artifactId>
-                <version>3.6.0</version>
+                <version>3.6.1</version>
                 <executions>
                     <execution>
                         <id>copy-jacoco</id>
@@ -301,10 +301,9 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-failsafe-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.2.2</version>
                 <configuration>
-                    <!-- Set the highest log level for coverage testing, so that we have a chance to reach branches
-          <property>	                      in the logging lambdas too. -->
+                    <!-- Set the highest log level for coverage testing, so that we have a chance to reach branches in the logging lambdas too. -->
                     <argLine>-Djava.util.logging.config.file=src/test/resources/logging.properties ${argLine}</argLine>
                     <systemPropertyVariables>
                         <test.coverage>true</test.coverage>

--- a/project-keeper-maven-plugin/pk_generated_parent.pom
+++ b/project-keeper-maven-plugin/pk_generated_parent.pom
@@ -71,6 +71,12 @@
                 <configuration>
                     <source>${java.version}</source>
                     <target>${java.version}</target>
+                    <showWarnings>true</showWarnings>
+                    <compilerArgs>
+                        <!-- We turn off `processing` because it complains about every annotation -->
+                        <arg>-Xlint:all,-processing</arg>
+                        <!-- Don't add -Werror as this will let the build fail for linter warnings -->
+                    </compilerArgs>
                 </configuration>
             </plugin>
             <plugin>

--- a/project-keeper-maven-plugin/pom.xml
+++ b/project-keeper-maven-plugin/pom.xml
@@ -87,7 +87,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-plugin-plugin</artifactId>
-                <version>3.10.1</version>
+                <version>3.10.2</version>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

--- a/project-keeper/pk_generated_parent.pom
+++ b/project-keeper/pk_generated_parent.pom
@@ -133,7 +133,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.2.2</version>
                 <configuration>
                     <!-- Set the highest log level for coverage testing, so that we
           have a chance to reach branches in the logging lambdas too. -->
@@ -231,7 +231,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.6.0</version>
+                <version>3.6.2</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>
@@ -273,10 +273,9 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-failsafe-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.2.2</version>
                 <configuration>
-                    <!-- Set the highest log level for coverage testing, so that we have a chance to reach branches
-          <property>	                      in the logging lambdas too. -->
+                    <!-- Set the highest log level for coverage testing, so that we have a chance to reach branches in the logging lambdas too. -->
                     <argLine>-Djava.util.logging.config.file=src/test/resources/logging.properties ${argLine}</argLine>
                     <systemPropertyVariables/>
                     <excludedGroups>${test.excludeTags}</excludedGroups>

--- a/project-keeper/pk_generated_parent.pom
+++ b/project-keeper/pk_generated_parent.pom
@@ -63,6 +63,12 @@
                 <configuration>
                     <source>${java.version}</source>
                     <target>${java.version}</target>
+                    <showWarnings>true</showWarnings>
+                    <compilerArgs>
+                        <!-- We turn off `processing` because it complains about every annotation -->
+                        <arg>-Xlint:all,-processing</arg>
+                        <!-- Don't add -Werror as this will let the build fail for linter warnings -->
+                    </compilerArgs>
                 </configuration>
             </plugin>
             <plugin>

--- a/project-keeper/src/main/resources/maven_templates/maven-compiler-plugin.xml
+++ b/project-keeper/src/main/resources/maven_templates/maven-compiler-plugin.xml
@@ -5,5 +5,11 @@
     <configuration>
         <source>${java.version}</source>
         <target>${java.version}</target>
+        <showWarnings>true</showWarnings>
+        <compilerArgs>
+            <!-- We turn off `processing` because it complains about every annotation -->
+            <arg>-Xlint:all,-processing</arg>
+            <!-- Don't add -Werror as this will let the build fail for linter warnings -->
+        </compilerArgs>
     </configuration>
 </plugin>

--- a/project-keeper/src/main/resources/maven_templates/maven-dependency-plugin.xml
+++ b/project-keeper/src/main/resources/maven_templates/maven-dependency-plugin.xml
@@ -1,7 +1,7 @@
 <plugin>
     <groupId>org.apache.maven.plugins</groupId>
     <artifactId>maven-dependency-plugin</artifactId>
-    <version>3.6.0</version>
+    <version>3.6.1</version>
     <executions>
         <execution>
             <id>copy-jacoco</id>

--- a/project-keeper/src/main/resources/maven_templates/maven-failsafe-plugin.xml
+++ b/project-keeper/src/main/resources/maven_templates/maven-failsafe-plugin.xml
@@ -1,10 +1,9 @@
 <plugin>
     <groupId>org.apache.maven.plugins</groupId>
     <artifactId>maven-failsafe-plugin</artifactId>
-    <version>3.1.2</version>
+    <version>3.2.2</version>
     <configuration>
-        <!-- Set the highest log level for coverage testing, so that we have a chance to reach branches
-          <property>	                      in the logging lambdas too. -->
+        <!-- Set the highest log level for coverage testing, so that we have a chance to reach branches in the logging lambdas too. -->
         <argLine>-Djava.util.logging.config.file=src/test/resources/logging.properties ${argLine}</argLine>
         <systemPropertyVariables>
             <test.coverage>true</test.coverage>

--- a/project-keeper/src/main/resources/maven_templates/maven-javadoc-plugin.xml
+++ b/project-keeper/src/main/resources/maven_templates/maven-javadoc-plugin.xml
@@ -1,7 +1,7 @@
 <plugin>
     <groupId>org.apache.maven.plugins</groupId>
     <artifactId>maven-javadoc-plugin</artifactId>
-    <version>3.6.0</version>
+    <version>3.6.2</version>
     <executions>
         <execution>
             <id>attach-javadocs</id>

--- a/project-keeper/src/main/resources/maven_templates/maven-surefire-plugin.xml
+++ b/project-keeper/src/main/resources/maven_templates/maven-surefire-plugin.xml
@@ -1,7 +1,7 @@
 <plugin>
     <groupId>org.apache.maven.plugins</groupId>
     <artifactId>maven-surefire-plugin</artifactId>
-    <version>3.1.2</version>
+    <version>3.2.2</version>
     <configuration>
         <!-- Set the highest log level for coverage testing, so that we
           have a chance to reach branches in the logging lambdas too. -->

--- a/shared-model-classes/pk_generated_parent.pom
+++ b/shared-model-classes/pk_generated_parent.pom
@@ -63,6 +63,12 @@
                 <configuration>
                     <source>${java.version}</source>
                     <target>${java.version}</target>
+                    <showWarnings>true</showWarnings>
+                    <compilerArgs>
+                        <!-- We turn off `processing` because it complains about every annotation -->
+                        <arg>-Xlint:all,-processing</arg>
+                        <!-- Don't add -Werror as this will let the build fail for linter warnings -->
+                    </compilerArgs>
                 </configuration>
             </plugin>
             <plugin>

--- a/shared-model-classes/pk_generated_parent.pom
+++ b/shared-model-classes/pk_generated_parent.pom
@@ -133,7 +133,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.2.2</version>
                 <configuration>
                     <!-- Set the highest log level for coverage testing, so that we
           have a chance to reach branches in the logging lambdas too. -->
@@ -231,7 +231,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.6.0</version>
+                <version>3.6.2</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>

--- a/shared-test-setup/pk_generated_parent.pom
+++ b/shared-test-setup/pk_generated_parent.pom
@@ -52,6 +52,12 @@
                 <configuration>
                     <source>${java.version}</source>
                     <target>${java.version}</target>
+                    <showWarnings>true</showWarnings>
+                    <compilerArgs>
+                        <!-- We turn off `processing` because it complains about every annotation -->
+                        <arg>-Xlint:all,-processing</arg>
+                        <!-- Don't add -Werror as this will let the build fail for linter warnings -->
+                    </compilerArgs>
                 </configuration>
             </plugin>
             <plugin>

--- a/shared-test-setup/pk_generated_parent.pom
+++ b/shared-test-setup/pk_generated_parent.pom
@@ -122,7 +122,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>3.1.2</version>
+                <version>3.2.2</version>
                 <configuration>
                     <!-- Set the highest log level for coverage testing, so that we
           have a chance to reach branches in the logging lambdas too. -->

